tensorflow_addons/layers/esn.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements Echo State recurrent Network (ESN) layer."""
+
+import tensorflow as tf
+from tensorflow_addons.rnn import ESNCell
+from typeguard import typechecked
+
+from tensorflow_addons.utils.types import (
+    Activation,
+    FloatTensorLike,
+    TensorLike,
+    Initializer,
+)
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ESN(tf.keras.layers.RNN):
+    """Echo State Network layer.
+
+    This implements the recurrent layer using the ESNCell.
+
+    This is based on the paper
+        H. Jaeger
+        "The "echo state" approach to analysing and training recurrent neural networks".
+        GMD Report148, German National Research Center for Information Technology, 2001.
+        https://www.researchgate.net/publication/215385037
```

Comment:
Could we use the hyperlink here? Like [xxx](link).

Reply:
@WindQAQ Thanks for your review :)

Do you mean something like this? 
_Option A:_
 ```
This is based on the paper
H. Jaeger
["The "echo state" approach to analysing and training recurrent neural networks".
GMD Report148, German National Research Center for Information Technology, 2001.]
(https://www.researchgate.net/publication/215385037)
```
_Option B:_
 ```
This is based on the [paper](https://www.researchgate.net/publication/215385037)
H. Jaeger
"The "echo state" approach to analysing and training recurrent neural networks".
GMD Report148, German National Research Center for Information Technology, 2001.
```
_Option C:_
 ```
This is based on the paper
H. Jaeger
["The "echo state" approach to analysing and training recurrent neural networks"]
(https://www.researchgate.net/publication/215385037).
GMD Report148, German National Research Center for Information Technology, 2001.
```

I think I prefer option C, please let me know your opinion

Reply:
@pedrolarben I prefer C too :-)

Reply:
Done! :+1: